tests/fixtures/fixtures.cpp

```diff
@@ -267,4 +276,30 @@ GenTestItems(uint64_t count, uint64_t max_length, uint64_t max_index) {
     return result;
 }
 
+template <typename T>
+static T*
+CopyVector(const std::vector<T>& vec) {
+    auto result = new T[vec.size()];
+    memcpy(result, vec.data(), vec.size() * sizeof(T));
+    return result;
+}
+
+vsag::DatasetPtr
+generate_one_dataset(int64_t dim, uint64_t count) {
+    auto result = vsag::Dataset::Make();
+    auto [ids, vectors] = generate_ids_and_vectors(count, dim, true, time(nullptr));
+    result->Dim(dim)
+        ->NumElements(count)
+        ->Float32Vectors(CopyVector(vectors))
+        ->Ids(CopyVector(ids))
```

Comment:
why we need copy vectors and ids here

Reply:
owner = true, make the lifetime manage by dataset